### ISSUE TITLE
👾 [Bug Report] Creature AIUpdate

### ISSUE THREAD

**Issue by aaron02** [AscEmu/AscEmu#816]
**Description**: 
The AiUpdate begins before you enter combat with the creature.
When you leave the battle and enter it again the aiupdate dont triggers.

**AscEmu hash/commit**: 
d5506b6

**Operating system**: 
    - Platform: Windows

**Comment by aaron02** [comment]
![image](https://user-images.githubusercontent.com/1266630/98469449-f7939500-21df-11eb-8348-a56a11dc9691.png)
with this it will work but the old timers are disabled then

**Comment by Appled** [comment]
Confirmed while testing Gruul. If you die during the encounter and try the encounter again he won't use spells anymore. Aaron says the ai is no longer updated. @Zyres ?

**Comment by Zyres** [comment]
> Confirmed while testing Gruul. If you die during the encounter and try the encounter again he won't use spells anymore. Aaron says the ai is no longer updated. @Zyres ?

As the comment states in CreatureAIScript Line 105: https://github.com/AscEmu/AscEmu/blob/master/src/world/Server/Script/CreatureAIScript.cpp#L105
``old AIUpdate stuff is now handled by customAIUpdateTimer. Keep this until all scripts are updated to new logic.``

Yes if we removed Line 106-112 the old timers won't work anymore. If we use "old timers" in new scripts we wont be able to update the script. Scripts using the old timers require some extra stuff in the script (like Registering AIUpdateEvents.
For reference check out this: https://github.com/AscEmu/AscEmu/blob/master/src/scripts/EventScripts/Event_L70ETC_Concert.cpp#L50-L68

Over all this is issue is based on the fact that we still support old timers (and scripts) and try to create a new logic within it.

Long story short, here some solutions:
1. Remove the old timer logic and update old scripts to new logic
2. Register AIUpdateEvents in OnCombatStart directly in the script and downgrade new scripts to old logic.
3. Mess up the new logic even more and add AUUpdateEvent in _internalOnCombatStart, so all new scripts use new and old timers and wait for the moment the logics are fucked up.

At this state I would personally prefer the first option :)
